scripts/configuration/AppCenterModule.cake

```diff
@@ -0,0 +1,45 @@
+// AppCenter module class definition.
+class AppCenterModule
+{
+    public string DotNetModule { get; }
+    public string NuGetVersion { get; }
+    public string MainNuspecFilename { get; }
+    public string NuGetPackageName => $"{_packageId}.{NuGetVersion}.nupkg";
+    public string MacNuspecFilename =>  "Mac" + MainNuspecFilename;
+    public string WindowsNuspecFilename => "Windows" + MainNuspecFilename;
+    private string _packageId;
+
+    private AppCenterModule(string dotnet, string mainNuspecFilename) {
+        DotNetModule = dotnet;
+        MainNuspecFilename = mainNuspecFilename;
+    }
+
+    private AppCenterModule(XmlNode moduleNode, string nuspecFolder, string nugetVersion)
+    {
+        DotNetModule = moduleNode.Attributes.GetNamedItem("dotnetModule").Value;
+        MainNuspecFilename = moduleNode.Attributes.GetNamedItem("nuspec").Value;
+        NuGetVersion = nugetVersion;
+        var nuspecText = Statics.Context.FileReadText(System.IO.Path.Combine(nuspecFolder, MainNuspecFilename));
+        var startTag = "<id>";
+        var endTag = "</id>";
+        int startIndex = nuspecText.IndexOf(startTag) + startTag.Length;
```

Comment:
Why not using xml apis then?

Reply:
This is older code that i just moved, but for now I think it's probably simple enough to keep. (We do similar things in version.cake)

Reply:
But they're used elsewhere in this file.... I would favor consistency within a file over consistency across files